pkg/parse/listener_impl.go

```diff
@@ -802,31 +805,33 @@ func (s *TreeShapeListener) EnterUnion(ctx *parser.UnionContext) {
 	type1.SourceContext = s.sc.Get(ctx.BaseParserRuleContext)
 }
 
+// EnterUnion_type is called when production union_type is entered.
+func (s *TreeShapeListener) EnterUnion_type(ctx *parser.Union_typeContext) {
+	fieldName := ctx.GetText()
+	s.fieldname = append(s.fieldname, fieldName)
+	dataType, has := s.typemap[fieldName]
+	if has {
+		logrus.Warnf("%s) %s.(%s) defined multiple times, merge definitions",
```

Comment:
This error could be confused to mean that the definitions are merged. I think it actually means that the second definition is ignored. Perhaps something like `%s) %s.(%s) defined multiple times, redefinition ignored` would be better.